src/in_memory_maintenance.rs

```diff
@@ -28,6 +33,9 @@ pub async fn in_memory_database_maintaince(
                     }
                 },
             };
+            if readiness_check(dune_download_folder.clone() + "user_data/")? {
```

Comment:
Maybe short circuit if the service is already ready?

```suggestion
            if !health.is_ready() && readiness_check(dune_download_folder.clone() + "user_data/")? {
```

Otherwise, if we want to check on every loop, we need to `health.notify_not_ready()` if the readiness check fails right?